libraries/provider_configure.rb

```diff
@@ -0,0 +1,91 @@
+class Chef
+  # Chef Provider for configuring an elasticsearch instance
+  class Provider::ElasticsearchConfigure < Provider
+    include Poise
+
+    def action_manage
+      # Create ES directories
+      #
+      [ new_resource.path_conf, new_resource.path_logs ].each do |path|
+        directory path do
+          owner new_resource.user
+          group new_resource.group
+          mode 0755
+          recursive true
+          action :create
+        end
+      end
+
+      # Create data path directories
+      #
+      data_paths = new_resource.path_data.is_a?(Array) ? new_resource.path_data : new_resource.path_data.split(',')
+
```

Comment:
That would be great -- let's strive for as much "separation" and modularity with the refactored cookbook as possible. I think my general approch is: we can save on implementing features, but invest into the codebase itself (modularity, testability, ...)